web/profiles/joinup/src/Plugin/Block/AddContentBlock.php

```diff
@@ -98,25 +98,27 @@ public function build() {
         'rdf_entity' => $collection_contexts['collection']->getContextValue()
           ->id(),
       ]);
-      $links['custom_page'] = [
-        '#type' => 'link',
-        '#title' => $this->t('Add custom page'),
-        '#url' => $page_url,
-        '#attributes' => ['class' => ['button', 'button--small']],
-        '#access' => $page_url->access(),
-      ];
+      if ($page_url->access()) {
+        $links['custom_page'] = [
+          '#type' => 'link',
+          '#title' => $this->t('Add custom page'),
+          '#url' => $page_url,
+          '#attributes' => ['class' => ['button', 'button--small']],
+        ];
+      }
```

Comment:
Little tip: if you want you can do this smarter :)

You can do something like this after compiling the links:

```
$links = array_filter($links, function ($link) {
  return $link['#url']->access();
});
```